tests/annotations_test.go

```diff
@@ -0,0 +1,84 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Ensures services.annotations values and the individual annotations per object
+// are applied correctly.
+func TestAnnotations(t *testing.T) {
+	t.Parallel()
+
+	var (
+		chart = LoadChart(t)
+
+		expectedGlobal = map[string]string{
+			"global-key": "global-value",
+			// Should be overwritten by some children.
+			"key": "global-value",
+		}
+		expectedCoderd = map[string]string{
+			"key":  "value",
+			"key2": "value2",
+		}
+		expectedCoderdService = map[string]string{
+			"key": "value",
+			"service.beta.kubernetes.io/aws-load-balancer-backend-protocol": "http",
+		}
+		expectedTimescale = map[string]string{
+			"key2": "value",
+		}
+	)
+
+	objs := chart.MustRender(t, func(cv *CoderValues) {
+		// Ensure backwards compatibility and merging order.
+		cv.Services.Annotations = expectedGlobal
+
+		cv.Coderd.Annotations = expectedCoderd
+		cv.Coderd.ServiceSpec.Annotations = expectedCoderdService
+		cv.Postgres.Default.Annotations = expectedTimescale
+	})
+
+	depl := MustFindDeployment(t, objs, "coderd")
+	assert.Equal(t, mergeAnnotations(expectedGlobal, expectedCoderd), depl.Annotations)
+
+	svc := MustFindService(t, objs, "coderd")
+	assert.Equal(t, mergeAnnotations(expectedGlobal, expectedCoderdService), svc.Annotations)
+
+	db := MustFindStatefulSet(t, objs, "timescale")
+	assert.Equal(t, mergeAnnotations(expectedGlobal, expectedTimescale), db.Annotations)
+}
+
+func TestAnnotationsEmpty(t *testing.T) {
+	t.Parallel()
+
+	var (
+		chart = LoadChart(t)
+		objs  = chart.MustRender(t, nil)
```

Comment:
you need a MustRender function that explicitly sets the annotation values to nil, you should be able to change the chart back to how you had it before, see a test failure, add back the null checks, and see that it has been resolved